CONTRIBUTING.md

```diff
@@ -2,35 +2,70 @@
+-   create or contribute to [integrations](docs/user-guide/integrations/editor.md), like our plugin for [VS Code](https://github.com/stylelint/vscode-stylelint)
 
-Our [vision](docs/about/vision.md) guides our work.
+**We want to do everything we can to encourage contributions!** So if you want to participate but don't end up doing it for one reason or another, please [file an issue](https://github.com/stylelint/stylelint/issues/new) and give us feedback about what we could do to encourage you better.
 
-**We want to do everything we can to encourage contributions!** So if you want to participate but don't end up doing it for one reason or another, please file an issue and give us feedback about what we could do to better encourage you.
+## Code contributions
 
-Also: we hope that your participation in the project isn't a one-off. *We'd love to add more members to the organization and see more regulars pop up in issues and pull requests!*
+You should first [create a new issue](https://github.com/stylelint/stylelint/issues/new/choose) so that we can discuss your new feature, bug fix or change.
 
-## Code contributions
+To start coding, you'll need:
+
+-   a minimum of [Node.js](https://nodejs.org/en/) v10, though we do recommend using the latest LTS release
+-   the latest [npm](https://www.npmjs.com/)
+
+Then:
+
+1.  [Fork and clone](https://guides.github.com/activities/forking/) the stylelint repository.
+2.  Install all the dependencies with `npm install`.
+
+### Run tests
+
+Next, you'll want to run the tests using `npm test`.
+
+However, this runs all 25,000+ unit tests and also linting.
+
+You can use the interactive testing prompt to run tests for just a chosen set of files (which you'll want to do during development). For example, to run the tests for just the `color-hex-case` and `color-hex-length` rules:
 
-You'll need:
+1.  Run `npm run watch` to start the interactive testing prompt.
+2.  Press `p` to filter by a filename regex pattern.
+3.  Enter `color-hex-case|color-hex-length` i.e. each rule name separated by the pipe symbol (`|`).
 
--   Minimum [Node.js](https://nodejs.org/en/) v10, though we do recommend using the latest LTS release.
--   Latest [npm](https://www.npmjs.com/).
+### Write code
 
-The [Developer guide](docs/toc.md) outlines how to get started contributing code.
+With the interactive testing prompt running, you can write the code to fix the bug or add the new feature confident that things are working as expected. You'll find more guidance on how we write code in the [Developer guide](docs/toc.md).
+
+### Format code
+
+After writing your code, you'll want to format it correctly. We use [prettier](https://prettier.io/) to do this. You can pretty-print all the files using `npm run prettier:fix`.
+
+Alternatively, you can use an [editor integration](https://prettier.io/docs/en/editors.html).
```

Comment:
[question] Do we need to mention to the auto-formatting via a pre-commit hook (`husky` & `lint-staged`)?